File: TTStest/app/src/main/java/com/example/ttstest/MainActivity.java
package com.example.ttstest;

import android.content.Intent;
import android.os.Bundle;
import android.speech.tts.TextToSpeech;
import android.speech.tts.TextToSpeech.OnInitListener;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import java.util.Locale;

public class MainActivity extends AppCompatActivity implements OnInitListener {

    private TextToSpeech tts;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        checkTTSData();
        // 初始化TextToSpeech
        tts = new TextToSpeech(this, this);
    }

    @Override
    public void onInit(int status) {
        if (status == TextToSpeech.SUCCESS) {
            // 设置语言
            int result = tts.setLanguage(Locale.CHINA);

            if (result == TextToSpeech.LANG_MISSING_DATA || result == TextToSpeech.LANG_NOT_SUPPORTED) {
                // 语言数据缺失或不支持
                System.out.println("Language is not supported");
            } else {
                // 播报文本
                speak("你有新的订单");
            }
        } else {
            Log.d("aaaa","Initialization Failed!");
        }
    }
    private void checkTTSData() {
        Intent checkIntent = new Intent();
        checkIntent.setAction(TextToSpeech.Engine.ACTION_CHECK_TTS_DATA);
        startActivityForResult(checkIntent, 0);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == 0) {
            if (resultCode == TextToSpeech.Engine.CHECK_VOICE_DATA_PASS) {
                // 设备已安装TTS引擎并且数据有效
                tts = new TextToSpeech(this, this);
            } else {
                // 设备未安装TTS引擎或数据无效，提示用户安装
                Intent installIntent = new Intent();
                installIntent.setAction(TextToSpeech.Engine.ACTION_INSTALL_TTS_DATA);
                startActivity(installIntent);
            }
        }
    }
    private void speak(String text) {
        tts.speak(text, TextToSpeech.QUEUE_FLUSH, null, null);
    }

    @Override
    protected void onDestroy() {
        // 关闭TTS引擎
        if (tts != null) {
            tts.stop();
            tts.shutdown();
        }
        super.onDestroy();
    }
}

File: FragmentTest/app/src/main/java/com/example/fragmenttest/MainActivity.kt
package com.example.fragmenttest

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.fragment.app.Fragment

class MainActivity : AppCompatActivity() {
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val btn:Button = findViewById(R.id.button)
        btn.setOnClickListener {
            replaceFragment(right_Fragment())
        }
      //  replaceFragment(right_Fragment())
 //       val Left_Fragment = supportFragmentManager.findFragmentById(R.id.rightFrag) as Left_Fragment
    }
    private  fun replaceFragment(fragment: Fragment) {
        val fragmentStateManager = supportFragmentManager
        val transition = fragmentStateManager.beginTransaction()
        right_Fragment.conut++
        transition.replace(R.id.rightFrag,fragment)
       // transition.addToBackStack(null)
        transition.commit()
    }
}